src/libopensc/card-esteid2018.c

```diff
@@ -114,53 +101,41 @@ static int esteid_select_file(struct sc_card *card, const struct sc_path *in_pat
 		LOG_FUNC_RETURN(card->ctx, SC_ERROR_INVALID_ARGUMENTS);
 	}
 
-	memcpy(path, in_path->value, in_path->len);
-	pathlen = in_path->len;
-
-	while (pathlen >= 2) {
+	for (size_t pathlen = in_path->len; pathlen >= 2; pathlen -= 2, path += 2) {
```

Comment:
maybe you want to error if pathlen is odd (currently the last byte is ignored)

Reply:
Fixed